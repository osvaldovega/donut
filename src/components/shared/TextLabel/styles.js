import { StyleSheet } from 'react-native';
import Colors from '../../../constants/Colors';

const styles = StyleSheet.create({
  wrapper: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  content: {
    fontSize: 12,
    color: Colors.goldenSand,
    fontFamily: 'montserrat',
    fontWeight: '600',
    textTransform: 'uppercase',
  },
});

export default styles;
