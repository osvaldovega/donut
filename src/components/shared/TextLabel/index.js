import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';

const TextLabel = ({ content, textStyle, containerStyle }) => (
  <View style={[styles.wrapper, containerStyle]}>
    <Text style={[styles.content, textStyle]}>{content}</Text>
  </View>
);

TextLabel.defaultProps = {
  containerStyle: '',
  textStyle: '',
  title: '',
};

export default TextLabel;
