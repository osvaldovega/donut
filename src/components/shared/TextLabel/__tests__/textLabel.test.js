import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import TextLabel from '../index';

describe('<TextLabel />', () => {

  const setup = (props = null) =>  {
    return props
    ? renderer.create(<TextLabel {...props} />)
    : renderer.create(<TextLabel />)
  };

  it('renders correctly', () => {
    const instance = setup();
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a prop value', () => {
    const newProps = { content: 'CoinBase' };
    const instance = setup(newProps);
    const textIns = instance.root.findByType(Text);
    expect(textIns.props.children).toBe(newProps.content);
  });
});
