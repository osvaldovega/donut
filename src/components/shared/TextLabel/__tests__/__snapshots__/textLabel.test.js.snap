// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLabel /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "width": "100%",
      },
      "",
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#eccc68",
          "fontFamily": "montserrat",
          "fontSize": 12,
          "fontWeight": "600",
          "textTransform": "uppercase",
        },
        "",
      ]
    }
  >
    
  </Text>
</View>
`;
